pkg/inputs/snmp/x/meraki/meraki.go

```diff
@@ -319,13 +329,164 @@ func (c *MerakiClient) Run(ctx context.Context, dur time.Duration) {
 				}
 			}
 
+			if doNetworkApplianceSecurityEvents {
+				if err := c.getNetworkApplianceSecurityEvents(dur); err != nil {
+					c.log.Infof("Meraki cannot get network security events: %v", err)
+				}
+			}
+
 		case <-ctx.Done():
 			c.log.Infof("Meraki Poll Done")
 			return
 		}
 	}
 }
 
+func (c *MerakiClient) getNetworkApplianceSecurityEvents(dur time.Duration) error {
+	startTime := time.Now().Add(-1 * dur)
+	startTimeStr := fmt.Sprintf("%v", startTime.Unix())
+
+	var getNetworkEvents func(nextToken string, network networkDesc) error
+	getNetworkEvents = func(nextToken string, network networkDesc) error {
+		params := appliance.NewGetNetworkApplianceSecurityEventsParamsWithTimeout(c.timeout)
+		params.SetNetworkID(network.ID)
+		params.SetT0(&startTimeStr)
+		if nextToken != "" {
+			params.SetStartingAfter(&nextToken)
+		}
+
+		prod, err := c.client.Appliance.GetNetworkApplianceSecurityEvents(params, c.auth)
+		if err != nil {
+			if strings.Contains(err.Error(), "(status 400)") { // There are no valid logs to worry about here.
+				return nil
+			}
+			return err
+		}
+
+		results := prod.GetPayload()
+		b, err := json.Marshal(results)
+		if err != nil {
+			return err
+		}
+		c.logchan <- string(b)
+
+		nextLink := getNextLink(prod.Link)
+		if nextLink != "" {
+			return getNetworkEvents(nextLink, network)
+		} else {
+			return nil
+		}
+	}
+
+	for _, org := range c.orgs {
+		for _, network := range org.networks {
+			err := getNetworkEvents("", network)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (c *MerakiClient) getNetworkEventsWrapper(ctx context.Context) {
+	c.log.Infof("Network Events Check Starting")
+	logCheck := time.NewTicker(1 * time.Hour)
+
+	nextPageEndAt := "" // Start from the very beginning.
+
+	// Check once on startup.
+	err, np := c.getNetworkEvents(nextPageEndAt)
+	if err != nil {
+		c.log.Errorf("Cannot get network events: %v", err)
+	} else {
+		nextPageEndAt = np
+	}
```

Comment:
could you use the [setPageStartAt](https://github.com/meraki/dashboard-api-go/blob/main/client/docs/GetNetworkEvents200Response.md#setpagestartat) parameter to create a poll interval? 

grabbing the previous [getPageEndAt](https://github.com/meraki/dashboard-api-go/blob/main/client/docs/GetNetworkEvents200Response.md#getpageendat) value to reference on the next polling cycle?

Reply:
I'm a little leary about doing that because I don't see it in the docs here: https://developer.cisco.com/meraki/api/get-network-events/

You're looking at the responce body object but if you look at the request one there's no easy place to put the startAt value.

Reply:
i think the idea would be to use it as a query param: 
<img width="1201" alt="Get Network Events - Meraki Dashboard API v1 - Cisco Meraki Developer Hub 2024-02-20 at 4 40 44 PM" src="https://github.com/kentik/ktranslate/assets/30831476/27374a5c-9913-4367-a55f-b9861381ec12">

Reply:
Isn't that what I'm doing here? https://github.com/kentik/ktranslate/pull/668/files#diff-5fec11c4942216760fa250830a33713575b00e08ab3fb77527165a55ae693c32R356

This gets the latest timestamp and then will use this for the `startingAfter` param on the next call. Only problem is that the first time the value is empty so you get everything from the last week.

Reply:
d'oh! ok, let's add this to the staging image and see what it looks like; I think there MIGHT be a use case in the future to add a runtime option for how many minutes to look back on start; but until there's a customer demand I don't think we should worry. There's obviously a risk of duplicate data during an agent upgrade, but it should not be super impactful IMO.